include/deal.II/fe/fe_poly.templates.h

```diff
@@ -472,43 +457,75 @@ FE_Poly<PolynomialType, dim, spacedim>::fill_fe_subface_values(
                           make_array_view(output_data.shape_3rd_derivatives,
                                           k));
 
-      for (unsigned int k = 0; k < this->dofs_per_cell; ++k)
-        correct_third_derivatives(output_data,
-                                  mapping_data,
-                                  quadrature.size(),
-                                  k);
+      correct_third_derivatives(output_data,
+                                mapping_data,
+                                quadrature.size(),
+                                mapping);
     }
 }
 
 
 
+template <class PolynomialType, int dim, int spacedim>
+inline void
+FE_Poly<PolynomialType, dim, spacedim>::correct_hessians(
+  internal::FEValuesImplementation::FiniteElementRelatedData<dim, spacedim>
+    &output_data,
+  const internal::FEValuesImplementation::MappingRelatedData<dim, spacedim>
+    &                           mapping_data,
+  const unsigned int            n_q_points,
+  const Mapping<dim, spacedim> &mapping) const
+{
+  // If we have a Cartesian mapping we don't need to correct the derivatives
+  // because the second and higher derivatives of the mapping are zero.
+  const bool need_to_correct_hessians =
+    nullptr == dynamic_cast<const MappingCartesian<dim> *>(&mapping);
```

Comment:
I think we typically place the constant on the right for comparison operations, i.e.,
```suggestion
    dynamic_cast<const MappingCartesian<dim> *>(&mapping) == nullptr;
```
However, I think we sometimes simply write
```c++
if (dynamic_cast<const MappingCartesian<dim> *>(&mapping))
  for ...
```